tests/ArrayHelper/ArrayHelperTest.php

```diff
@@ -119,11 +120,56 @@ public function testIsSubset(): void
         $this->assertFalse(ArrayHelper::isSubset(new ArrayObject([1]), ['1', 'b'], true));
     }
 
-    public function testGetObjectVars()
+    public function testGetObjectVars(): void
     {
         $this->assertSame([
             'id' => 123,
             'content' => 'test',
         ], ArrayHelper::getObjectVars(new Post2()));
     }
+
+    public function parsePathDataProvider(): array
+    {
+        return [
+            ['key1.key2.key3', '.', ['key1', 'key2', 'key3']],
+            ['key1\.key2.key3', '.', ['key1\.key2', 'key3']],
+            ['key1\:key2:key3', ':', ['key1\:key2', 'key3']],
```

Comment:
```suggestion
            ['key1\.key2.key3', '.', ['key1.key2', 'key3']],
            ['key1\:key2:key3', ':', ['key1:key2', 'key3']],
```

If `\` escape is dot, then result must be like this.